scripts/fuzz_opt.py

```diff
@@ -292,27 +292,80 @@ def count_runs(self):
 # Run VMs and compare results
 
 class VM:
-    def __init__(self, name, run, deterministic_nans, requires_legalization):
+    def __init__(self, name, run, can_run, can_compare_to_self, can_compare_to_others):
         self.name = name
         self.run = run
-        self.deterministic_nans = deterministic_nans
-        self.requires_legalization = requires_legalization
+        self.can_run = can_run
+        self.can_compare_to_self = can_compare_to_self
+        self.can_compare_to_others = can_compare_to_others
 
 
 class CompareVMs(TestCaseHandler):
     def __init__(self):
         super(CompareVMs, self).__init__()
 
-        def run_binaryen_interpreter(wasm):
+        def byn_run(wasm):
             return run_bynterp(wasm, ['--fuzz-exec-before'])
 
-        def run_v8(wasm):
+        def v8_run(wasm):
             run([in_bin('wasm-opt'), wasm, '--emit-js-wrapper=' + wasm + '.js'] + FEATURE_OPTS)
             return run_vm([shared.V8, wasm + '.js'] + shared.V8_OPTS + ['--', wasm])
 
+        def yes():
+            return True
+
+        def if_legal_and_no_nans():
+            return LEGALIZE and not NANS
+
+        def if_no_nans():
+            return not NANS
+
+        class Wasm2C(VM):
+            name = 'wasm2c'
+
+            def __init__(self):
+                # look for wabt in the path. if it's not here, don't run wasm2c
+                try:
+                    wabt_bin = run(['whereis', 'wasm2c'])
```

Comment:
`which` is probably much more suitable here.   Then you don't need to parse the output.

Actually, even better we have an existing `which()` function in `scripts/test/shared.py`